tc/core/compiler.cc

```diff
@@ -67,6 +67,25 @@ void checkInputsCompliant(
           << "expected a tensor with " << hdim << " dimensions but found "
           << dldim << " dimensions.";
     }
+    auto dlstrides = inputsInfo[i]->strides;
+    auto dlsizes = inputsInfo[i]->shape;
+    lang::ErrorReport ep =
```

Comment:
no reason to do this in the critical path, you prob want a helper function that throws and call it once you have detected the faulty case